azure/src/main/java/org/apache/iceberg/azure/AzureProperties.java

```diff
@@ -90,7 +117,9 @@ public Optional<Long> adlsWriteBlockSize() {
    */
   public void applyClientConfiguration(String account, DataLakeFileSystemClientBuilder builder) {
     String sasToken = adlsSasTokens.get(account);
-    if (sasToken != null && !sasToken.isEmpty()) {
+    if (adlsRefreshCredentialsEnabled && !Strings.isNullOrEmpty(adlsRefreshCredentialsEndpoint)) {
```

Comment:
I think here you could just check whether `vendedAdlsCredentialProvider` is not null

Reply:
Yes, that simplifies the check. Changed as per the suggestion.